src/backend/base/langflow/api/v1/chat.py

```diff
@@ -638,70 +723,6 @@ async def build_vertex_stream(
-                if not cache:
-                    # If there's no cache
-                    msg = f"No cache found for {flow_id_str}."
-                    raise ValueError(msg)
-                else:
-                    graph = cache.get("result")
-
-                vertex: InterfaceVertex = graph.get_vertex(vertex_id)
-                if not hasattr(vertex, "stream"):
-                    msg = f"Vertex {vertex_id} does not support streaming"
-                    raise ValueError(msg)
-                if isinstance(vertex._built_result, str) and vertex._built_result:
-                    stream_data = StreamData(
-                        event="message",
-                        data={"message": f"Streaming vertex {vertex_id}"},
-                    )
-                    yield str(stream_data)
-                    stream_data = StreamData(
-                        event="message",
-                        data={"chunk": vertex._built_result},
-                    )
-                    yield str(stream_data)
-
-                elif not vertex.frozen or not vertex._built:
-                    logger.debug(f"Streaming vertex {vertex_id}")
-                    stream_data = StreamData(
-                        event="message",
-                        data={"message": f"Streaming vertex {vertex_id}"},
-                    )
-                    yield str(stream_data)
-                    async for chunk in vertex.stream():
-                        stream_data = StreamData(
-                            event="message",
-                            data={"chunk": chunk},
-                        )
-                        yield str(stream_data)
-                elif vertex.result is not None:
-                    stream_data = StreamData(
-                        event="message",
-                        data={"chunk": vertex._built_result},
-                    )
-                    yield str(stream_data)
-                else:
-                    msg = f"No result found for vertex {vertex_id}"
-                    raise ValueError(msg)
-
-            except Exception as exc:  # noqa: BLE001
-                logger.exception("Error building Component")
-                exc_message = parse_exception(exc)
-                if exc_message == "The message must be an iterator or an async iterator.":
-                    exc_message = "This stream has already been closed."
-                yield str(StreamData(event="error", data={"error": exc_message}))
-            finally:
-                logger.debug("Closing stream")
-                if graph:
-                    await chat_service.set_cache(flow_id_str, graph)
-                yield str(StreamData(event="close", data={"message": "Stream closed"}))
-
-        return StreamingResponse(stream_vertex(), media_type="text/event-stream")
+        return StreamingResponse(_stream_vertex(str(flow_id), vertex_id, chat_service), media_type="text/event-stream")
```

Comment:
Were you able to test this? IIRC, it is a requirement to define the function inside the endpoint.

Reply:
I did a quick unit test and didn't see any problem.
`log_router.stream_logs` also defines its stream generator outside of its endpoint.

Reply:
Ran unit tests and it worked.